app/src/components/ConnectSubstrateWalletButton.tsx

```diff
@@ -0,0 +1,71 @@
+'use client'
+import { truncateAddress } from '@/utils/address'
+import Identicon from '@polkadot/react-identicon'
+import { WalletSelect } from '@talismn/connect-components'
+import { FC, useState } from 'react'
+
+interface WalletAccount {
+  address: string
+  source: string
+  name?: string
+  wallet?: unknown
+  signer?: unknown
+}
+
+interface ConnectSubstrateWalletButtonProps {
+  /** Text shown inside the button. */
+  label?: string
+}
+
+const ConnectSubstrateWalletButton: FC<ConnectSubstrateWalletButtonProps> = ({
+  label = 'Connect Wallet',
+}) => {
+  const [activeAccount, setActiveAccount] = useState<WalletAccount | null>(null)
+
+  // removes the active account if it is removed from the app
+  const handleUpdatedAccounts = (accounts?: WalletAccount[]) => {
+    if (!accounts || !activeAccount) return
+
+    for (let account of accounts) {
+      if (account.address === activeAccount.address) return
+    }
```

Comment:
this is fine but we could also `some` here:

```suggestion
    if (accounts.some(x => x.address === activeAccount.address)) { return }
```